fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/runners/throttled/ThrottledIteratorTest.java

```diff
@@ -644,8 +653,8 @@ void testIteratorClosesIncompleteFutures() throws Exception {
                 .withTransactionInitNotification(initNotification);
         try (ThrottledRetryingIterator<Integer> iterator = builder.build()) {
             iterateAll = iterator.iterateAll(storeBuilder);
-            // "sleep" for 10 ms to allow futures to get scheduled
-            MoreAsyncUtil.delayedFuture(10, TimeUnit.MILLISECONDS).join();
+            // Block here to let the futures a chance to get scheduled (the store future will trigger the iteration and then the itemHandler)
+            gate.acquire();
```

Comment:
There might still be some flakiness here, because I think you can
`release` and then `acquire` before you add to the `futures` and release.
But I don't think we can do much better.